### DISCUSSION TITLE
Ecosystem support for Markdown Strings?

### DISCUSSION THREAD

**Discussion by philCryoport** [mermaid-js/mermaid#5854] [Q&A] [answered]
I'm finding poor support for [Markdown Strings](https://mermaid.js.org/syntax/flowchart.html#markdown-strings) (aka "soft word-wrap") in Mermaid plug-ins / add-ons / integrations.

Paste this Mermaid flowchart code into any Mermaid plugin / add-on:

```
graph TD
    A["`Bacon ipsum dolor amet alc`"]
    B["`Bacon ipsum dolor amet alca`"]
```

The `A` block will display just fine most likely.  

So far I've only found THREE Mermaid add-on / plug-in / integration that'll display the text of the `B` block -- with a nice soft-wrap.  Two IDE plugins -- and the mermaid CLI 🤔 🤔 🤔

The rest of them just give a tiny empty-string block for `B` ...and that's it.

- [JetBrains Mermaid Plugin](https://plugins.jetbrains.com/plugin/20146-mermaid): PASS ✅
<img width="1200" alt="image" src="https://github.com/user-attachments/assets/81076b2d-7030-468b-982b-537c115e3e78">

- [VSCode Markdown Preview Mermaid Support](https://open-vsx.org/vscode/item?itemName=bierner.markdown-mermaid): PASS ✅
<img width="1346" alt="image" src="https://github.com/user-attachments/assets/be7a34dc-25ef-4dd5-9829-89af6175b734">

- Mermaid-CLI: PASS ✅
<img width="504" alt="image" src="https://github.com/user-attachments/assets/eef7373c-8fe9-4e6e-96fb-ae755b3322c2">

Everything else I've tried FAILS.  What's going on?  Where is the support for Markdown Strings?

- Mermaid Chart [Yes THAT Mermaid Chart:](https://mermaid.js.org/ecosystem/mermaid-chart.html) (with a longer string for `B`): FAIL ❌

<img width="1905" alt="image" src="https://github.com/user-attachments/assets/d0bd5eee-f16a-4e75-af34-ff73131a99c4">

 "[Mermaid Chart](https://www.mermaidchart.com/) was born out of the Mermaid open source project and was founded by Knut Sveidqvist together with Open Core Ventures. The lead developers from Mermaid have joined the company and there is a strong connection between the project we all love and Mermaid Chart. Mermaid Chart brings resources to the open source development of Mermaid and makes it possible to work with Mermaid professionally."

- [Mermaid.Live](https://mermaid.live): FAIL ❌
<img width="1908" alt="image" src="https://github.com/user-attachments/assets/1a890fc6-d3cf-410b-a00b-185823480c16">

- Github: FAIL ❌

```mermaid
graph TD
    A["`Bacon ipsum dolor amet alc`"]
    B["`Bacon ipsum dolor amet alca`"]
```

- [Atlassian Labs' Mermaid diagrams viewer for Confluence](https://marketplace.atlassian.com/apps/1232887/mermaid-diagrams-viewer?hosting=cloud&tab=overview): FAIL ❌

<img width="805" alt="image" src="https://github.com/user-attachments/assets/0036cf28-1be3-4c12-b342-f77f56937276">

- [Warsaw Dynamics's Mermaid Macro for Confluence](https://marketplace.atlassian.com/apps/1227600/plantuml-swagger-open-api-markdown-html-latex-asciidoc?hosting=cloud&tab=overview): FAIL ❌

<img width="813" alt="image" src="https://github.com/user-attachments/assets/357d6f4e-f8f9-417f-8dd9-06986a7f5146">

- Microsoft Loop's Mermaid integration: FAIL ❌

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/b777e6df-eb01-4e28-aa7e-3b69494222f8">

Why is the support for Markdown Strings so poor?  Not even THE OFFICIAL MERMAID CHART SERVICE supports it.

**Comment by sidharthv96** [comment]
Thanks for pointing it out! 

It's a recent bug, which is fixed in the latest version. We'll be rolling it out to Mermaid.live and MermaidChart soon. We're working on standardizing native markdown support soon, so markdown strings should be available in all diagrams in the same syntax, once the new renderer is out. 

Mermaid.live with [latest mermaid version](https://develop.git.mermaid.live/edit#pako:eNp9zj0KwzAMBeCrmDfnBN4acoN2qwoRthqb-ic49lBC7l6H7NWkB5942mGyFWgshVenHhMl1ef2JMwjm5yUX7cWlc0hF8VRquJgZsLrguN_yKfEgCglsre9Zz_vCNVJFILuq-XyIVA6uuNW8_2bDHQtTQaU3BYH_eaw9dRWy1Umz_3ZeJHjB9apQGw). 
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/acdc7037-690d-4ac2-a4ab-b84111917875">

GitHub seems to be working now?
<img width="931" alt="image" src="https://github.com/user-attachments/assets/0cb25b4d-8696-4248-8c59-659d7b20e84c">

**Reply by sidharthv96** [reply]
Just released live editor with latest mermaid. Please verify it is working.

**Reply by philCryoport** [reply]
> Just released live editor with latest mermaid. Please verify it is working.

Confirmed:
* mermaid.live `v11.2.0` [DOES _NOT_ work](https://mermaid.live/edit#pako:eNp9zj0KwzAMBeCrmDfnBN4acoN2qwoRtpqE-g_HHkLI3evivZr04BNPJ0y0Ao0lc1rVY6Kg2tyehHlkE4Pa0l69stHFrNhLUezMTHh1OP6H3CUGeMmeN9uazt8loazihaDbajl_CBSu5riWeD-CgS65yoAc67JCv9ntLdVkuci0cXvXd3J9AWMiQTY) 
_AND_
* mermaid.live `v11.2.0+d2de970` [DOES work](https://develop.git.mermaid.live/edit#pako:eNp9zj0KwzAMBeCrmDfnBN4acoN2qwoRthqb-ic49lBC7l6H7NWkB5942mGyFWgshVenHhMl1ef2JMwjm5yUX7cWlc0hF8VRquJgZsLrguN_yKfEgCglsre9Zz_vCNVJFILuq-XyIVA6uuNW8_2bDHQtTQaU3BYH_eaw9dRWy1Umz_3ZeJHjB9apQGw)

<img width="1476" alt="image" src="https://github.com/user-attachments/assets/f8f3e928-7e55-4fa1-b44b-0f83f8c76832">

**Reply by sidharthv96** [reply]
<img width="1308" alt="image" src="https://github.com/user-attachments/assets/e9f5deb7-1446-4ff5-823c-baedf6fd9b32">

Are you on firefox? We have an unreleased fix for firefox.

**Reply by sidharthv96** [reply, marked as answer]
That's released in v11.2.1

**Reply by philCryoport** [reply]
> Are you on firefox? We have an unreleased fix for firefox.

Indeed, I am a long-time Firefox user -- going back to the days of Netscape Navigator (yes, I'm old).

I just tried the standard [Mermaid.live link](https://mermaid.live/edit#pako:eNp9zj0KwzAMBeCrmDfnBN4acoN2qwoRtpqE-g_HHkLI3evivZr04BNPJ0y0Ao0lc1rVY6Kg2tyehHlkE4Pa0l69stHFrNhLUezMTHh1OP6H3CUGeMmeN9uazt8loazihaDbajl_CBSu5riWeD-CgS65yoAc67JCv9ntLdVkuci0cXvXd3J9AWMiQTY) -- now apparently running `v11.2.1` -- and it works within Firefox:

<img width="1466" alt="image" src="https://github.com/user-attachments/assets/f34ba5c1-bd47-477e-9e3a-80d9ccc31eb6">

**Reply by sidharthv96** [reply]
Rest of the plugins will start working once they upgrade to `v11.2.1`. You can check the current version by just typing `info`. 

```mermaid
info
```

**Comment by philCryoport** [comment]
@sidharthv96 wrote:
> GitHub seems to be working now? <img alt="image" width="931" src="https://private-user-images.githubusercontent.com/10703445/367654339-0cb25b4d-8696-4248-8c59-659d7b20e84c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1MDMxNzMsIm5iZiI6MTcyNjUwMjg3MywicGF0aCI6Ii8xMDcwMzQ0NS8zNjc2NTQzMzktMGNiMjViNGQtODY5Ni00MjQ4LThjNTktNjU5ZDdiMjBlODRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE2VDE2MDc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5NWExYzNiNzY4Y2U5ZDBlNzE1NGFiM2UxYjg4NzkwODhlZTkzNGY4ZjFmZmQ0ZjUxNjU5ODQ5NWEzZDc0YTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OPe3-AsjH1YxuQJ6hbf26lYmSP8l4dg1cBQgtRlww9s">

Trying within Github right here right now: FAIL ❌

Same code as documented initially above:
````
```mermaid
graph TD
    A["`Bacon ipsum dolor amet alc`"]
    B["`Bacon ipsum dolor amet alca`"]
```
````

Screenshot as of 2024-09-16T1709Z:
<img width="648" alt="image" src="https://github.com/user-attachments/assets/3634ea08-8f0f-44d8-8523-b7efaccf061e">

(FYI still broken as of 2024-09-16T2104Z)

Github doing it right here right now LIVE:
```mermaid
graph TD
    A["`Bacon ipsum dolor amet alc`"]
    B["`Bacon ipsum dolor amet alca`"]
```

**Reply by sidharthv96** [reply]
<img width="920" alt="image" src="https://github.com/user-attachments/assets/6f3d73a7-6c61-495c-b340-2928d732af82">
This should be fixed in firefox once GitHub upgrades to v11.2.1 or above.

**Reply by philCryoport** [reply]
Checking just now...Github works as expected:
![image](https://github.com/user-attachments/assets/dbbd82d9-802e-4554-9a88-86a7ea0d3329)
